### ISSUE TITLE
Automate documentation testing with a PR

### ISSUE THREAD

**Issue by duvivier** [CICE-Consortium/CICE#248]
We want to integrate test building of documentation for any PR that has modifications to the /doc/ directory (or just any pull request)? Possible ways of doing this include possibly using a probot or incorporating it into the Travis CI tests.

**Comment by phil-blain** [comment]
This is a planned feature of readthedocs, apparently coming this year !
[https://github.com/rtfd/readthedocs.org/issues/1340](https://github.com/rtfd/readthedocs.org/issues/1340)

**Comment by duvivier** [comment]
This is great news. It might just be worth the wait of holding out to see what they come up with instead of inventing the wheel ourselves.

**Comment by phil-blain** [comment]
See also the project board https://github.com/orgs/readthedocs/projects/8

Apparently the feature is ready to beta test:  https://blog.readthedocs.com/building-docs-for-pull-requests/ @duvivier if you want to try it out.

**Comment by apcraig** [comment]
We just received early access to the PR capability with readthedocs and I have turned on the webhook for  readthedocs in both Icepack and CICE to "pull request" as directed by readthedocs folks.  We'll have to see what happens.  My hope is with the next pull request, we get a new link in the PR pointing to the readthedocs documentation build.  We'll see.

**Comment by apcraig** [comment]
This seems to be working now in both Icepack and CICE.  I will close this for now.  If we need to reopen or need to open a new issue if something comes up, we can.

**Comment by phil-blain** [comment]
Just to document, what was missing to make it work ?

**Comment by apcraig** [comment]
This is a good question.  The final piece of the puzzle was to go into my personal settings, under applications, authorized OAuth apps, click on readthedocs, then make sure readthedocs was authorized in the CICE-Consortium.  

The permissions thing is something I still don't totally understand.  I have to do things in my personal settings under github and readthedocs to allow the CICE-Consortium Project to interact between github and readthedocs.  I guess this is because I have some admin permissions, but it would make more sense if I set something in the CICE-Consortium settings in github and readthedocs as an admin to set the permissions.

The other thing that had to be done was in the github CICE-Consortium Icepack repo settings, make sure the webhook for readthedocs had Pull Requests turned on.

Finally, something was done on the readthedocs side outside my control because this was a beta feature.  So it had to be turned on for the cice-consortium.  I assume once this is formally supported in readthedocs, it will be "on" for all projects.

As I said, it's working and I understand what I did.  But I'm still fuzzy how it all works.
